Source/ConnectionLostRetryPolicy.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  Copyright (c) 2014-2018 Alamofire Software Foundation (http://alamofire.org/)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// A retry policy that automatically retries idempotent requests for network connection lost errors. For more
+/// information about retrying network connection lost errors, please refer to Apple's
+/// [technical document](https://developer.apple.com/library/content/qa/qa1941/_index.html).
+public struct ConnectionLostRetryPolicy: RetryPolicy {
+
+    // MARK: - Properties
+
+    /// The total number of times the request is allowed to be retried.
+    public let retryLimit: UInt
+
+    /// The base of the exponential backoff policy (should always be greater than or equal to 2).
+    public let exponentialBackoffBase: UInt
+
+    /// The scale of the exponential backoff.
+    public let exponentialBackoffScale: Double
+
+    /// The idempotent http methods to retry.
+    /// See [RFC 2616 - Section 9.1.2](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html) for more information.
+    public let idempotentMethods: Set<HTTPMethod>
+
+    // MARK: - Initialization
+
+    /// Creates a `ConnectionLostRetryPolicy` instance from the specified parameters.
+    ///
+    /// - Parameters:
+    ///   - retryLimit:              The total number of times the request is allowed to be retried. `2` by default.
+    ///   - exponentialBackoffBase:  The base of the exponential backoff policy. `2` by default.
+    ///   - exponentialBackoffScale: The scale of the exponential backoff. `0.25` by default.
+    ///   - idempotentMethods:       The idempotent http methods to retry. `[.get, .head, .options, .trace]` by default.
+    public init(
+        retryLimit: UInt = 2,
+        exponentialBackoffBase: UInt = 2,
+        exponentialBackoffScale: Double = 0.25,
+        idempotentMethods: Set<HTTPMethod> = [.get, .head, .put, .delete, .options, .trace])
```

Comment:
At Nike, we don't use enable this policy for `.put` and `.delete` HTTP methods. We have some legacy services that are not idempotent in certain cases which makes it unsafe to simply retry the request. I figured the default Alamofire ones should follow the spec as closely as possible, and allow the methods to be customized.